packages/datadog-plugin-rhea/test/index.spec.js

```diff
@@ -0,0 +1,544 @@
+                  'amqp.link.role': 'sender',
+                  'amqp.delivery.state': 'accepted',
+                  'out.host': 'localhost',
+                  'out.port': '5673'
+                })
+              })
+                .then(done, done)
+              context.sender.send({ body: 'Hello World!' })
+            })
+
+            it('should inject span context', () => {
+              container.on('message', msg => {
+                const keys = Object.keys(msg.message.delivery_annotations)
+                expect(keys).to.include('x-datadog-trace-id')
+                expect(keys).to.include('x-datadog-parent-id')
+              })
+            })
+          })
+
+          describe('receiving a message', () => {
+            it('should automatically instrument', done => {
+              agent.use(traces => {
+                const span = traces[0][0]
+                expect(span).to.include({
+                  name: 'rhea.receiver.onmessage',
+                  resource: 'amq.topic',
+                  error: 0,
+                  service: 'test-amqp'
+                })
+                expect(span.meta).to.include({
+                  'span.kind': 'consumer',
+                  'amqp.link.source.address': 'amq.topic',
+                  'amqp.link.role': 'receiver'
+                })
+              })
+                .then(done, done)
+              context.sender.send({ body: 'Hello World!' })
+            })
+
+            it('should extract the span context', done => {
+              container.on('message', msg => {
+                const span = tracer.scope().active()
+                expect(span._spanContext._parentId).to.not.be.null
+                done()
+              })
+              context.sender.send({ body: 'Hello World!' })
+            })
+          })
+
+          describe('exception in message handler', () => {
+            it('span should have error', (done) => {
+              const Session = require(`../../../versions/rhea@${version}/node_modules/rhea/lib/session.js`)
+              const on_transfer = Session.prototype.on_transfer // eslint-disable-line camelcase
+              Session.prototype.on_transfer = function on_transfer_wrapped () { // eslint-disable-line camelcase
+                try {
+                  return on_transfer.apply(this, arguments)
+                } catch (e) {
+                  // this is just to prevent mocha from crashing
+                }
+              }
+
+              container.on('message', () => {
+                throw new Error('this is an error')
+              })
+
+              agent.use(traces => {
+                const span = traces[0][0]
+                expect(span.error).to.equal(1)
+                expect(span.meta).to.include({
+                  'error.msg': 'this is an error',
+                  'error.type': 'Error'
+                })
+                expect(span.meta['error.stack'].startsWith('Error: this is an error\n    at')).to.be.true
```

Comment:
It would be more robust to store the error in a variable and compare directly to the error properties.

Reply:
And in fact I'm doing exactly that in another test! Will fix.